sleap/io/format/nix.py

```diff
@@ -0,0 +1,387 @@
+
+        def create_feature_array(name, type, block, frame_index_array, shape, dtype):
+            array = block.create_data_array(name, type, dtype=dtype, shape=shape)
+            rd = array.append_range_dimension()
+            rd.link_data_array(frame_index_array, [-1])
+            return array
+
+        def create_positions_array(
+            name, type, block, frame_index_array, node_names, shape, dtype
+        ):
+            array = block.create_data_array(
+                name, type, dtype=dtype, shape=shape, label="pixel"
+            )
+            rd = array.append_range_dimension()
+            rd.link_data_array(frame_index_array, [-1])
+            array.append_set_dimension(["x", "y"])
+            array.append_set_dimension(node_names)
+            return array
+
+        def chunked_write(
+            instances,
+            frameid_array,
+            positions_array,
+            track_array,
+            skeleton_array,
+            pointscore_array,
+            instancescore_array,
+            trackingscore_array,
+            centroid_array,
+            track_map,
+            node_map,
+            skeleton_map,
+            chunksize=10000,
+        ):
+            data_written = 0
+            indices = np.zeros(chunksize, dtype=int)
+            positions = np.zeros((chunksize, 2, len(node_map.keys())), dtype=float)
+            centroids = np.zeros((chunksize, 2), dtype=float)
+            track = np.zeros_like(indices)
+            skeleton = np.zeros_like(indices)
+            pointscore = np.zeros((chunksize, len(node_map.keys())), dtype=float)
+            instscore = np.zeros_like(track, dtype=float)
+            trackscore = np.zeros_like(instscore)
+            dflt_pointscore = [0.0 for n in range(len(node_map.keys()))]
+
+            while data_written < len(instances):
+                print(".", end="")
+                start = data_written
+                end = (
+                    len(instances)
+                    if start + chunksize >= len(instances)
+                    else start + chunksize
+                )
+                for i in range(start, end):
+                    inst = instances[i]
+                    index = i - start
+                    indices[index] = inst.frame_idx
+                    if inst.track is not None:
+                        track[index] = track_map[inst.track.name]
+                    else:
+                        track[index] = -1
+                    skeleton[index] = skeleton_map[inst.skeleton.name]
+
+                    fnames = set([n.name for n in inst.nodes])
+                    nnames = set(list(node_map.keys()))
+                    missing = nnames.difference(fnames)
+
+                    for n, p in zip(inst.nodes, inst.points):
+                        positions[index, :, node_map[n.name]] = np.array([p.x, p.y])
+                    for m in missing:
+                        positions[index, :, node_map[m]] = np.array([np.nan, np.nan])
```

Comment:
Flagging this because bit of code assumes that all the nodes used by `inst` are listed in `node_map`. No action required, just ensure `node_map` continues to contain all `Node`s from any `Skeleton` used in the project.

Reply:
good catch! Since we now store all nodes whether or not they are part of a skeleton, this should no longer be an issue.